test/CheckedC/dynamic-checks/struct-increment-code-gen.c

```diff
@@ -109,7 +115,11 @@ void f1(void) {
   // CHECK-AST-NEXT: IntegerLiteral {{.*}} 2
 
   // CHECK-IR: [[REG1:%[a-zA-Z0-9.]*]] = load %struct.S1*, %struct.S1** @gp3
+  // NOTE: no check here, because we don't yet do checks at pointer arithmetic
```

Comment:
There's no non-null check here because `gp3 + 2` doesn't currently generate a non-null check of `gp3`. This is still an outstanding TODO, but will probably be clearer once we know where to insert the pointer arithmetic checks. 

I could look at adding them in this PR if you would prefer though.